src/plugins/URLIndicatorPlugin/URLIndicator.js

```diff
@@ -0,0 +1,103 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+define(
+    [],
+    function () {
+
+        // Set of connection states; changing among these states will be
+        // reflected in the indicator's appearance.
+        // CONNECTED: Everything nominal, expect to be able to read/write.
+        // DISCONNECTED: HTTP failed; maybe misconfigured, disconnected.
+        // PENDING: Still trying to connect, and haven't failed yet.
+        // ERROR: Got answer but it's due an internal server error or similar.
+        var CONNECTED = {
+                text: "connected",
+                glyphClass: "ok",
+                description: "Connected to the URL"
+            },
+            DISCONNECTED = {
+                text: "disconnected",
+                glyphClass: undefined,
+                description: "URL is offline"
+            },
+            PENDING = {
+                text: "Checking connection...",
+                glyphClass: 'caution',
+                description: "Unable to connect to the URL"
+            },
+            ERROR = {
+                text: "Error",
+                glyphClass: "err",
+                description: "Error trying to connect to the URL"
+            };
+        function URLIndicator($http, $interval) {
+            var self = this;
+            // Get the current connection state and options provided
+            this.icon = "icon-" + (this.options.icon ? this.options.icon : "database");
+            this.URLpath = this.options.url;
+            this.interval = this.options.interval || 1000;
+            this.state = PENDING;
+            if (this.options.label) {
+                this.label = this.options.label;
+            } else {
+                this.label = "URL";
+            }
+            // Callback if the HTTP request to URLIndicatorPlugin fails
+            function handleError(e) {
+                if (e.status !== 404) {
+                    self.state = ERROR;
+                } else {
+                    self.state = DISCONNECTED;
+                }
+            }
+            // Callback if the HTTP request succeeds.
+            function handleResponse() {
+                self.state = CONNECTED;
+            }
+            // Try to connect to the URL, and update the indicator.
+            function updateIndicator(path) {
+                path = self.URLpath;
+                $http.get(path).then(handleResponse, handleError);
```

Comment:
Please clean this up: replace `path` with `self.URLpath` to get rid of the path argument & variable entirely.